### PR TITLE
Fix: APP-1326 - Cached proposal showing on every DAO

### PR DESCRIPTION
**Pull request by Fabricevladimir**
## Description

- updates cached proposal key to fix issue of cached proposal being read on every DAO
- adds the cached votes to cached proposal as well

Task: [APP-1326](https://aragonassociation.atlassian.net/browse/APP-1326)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

File: packages/web-app/src/hooks/useDaoProposal.tsx [hunk 1]
```diff
@@ -38,18 +37,23 @@ export const useDaoProposal = (
   const cachedVotes = useReactiveVar(pendingVotesVar);
   const cachedProposals = useReactiveVar(pendingProposalsVar);
 
+  const cachedProposalId = useMemo(
+    () => generateCachedProposalId(daoAddress, proposalId),
+    [daoAddress, proposalId]
+  );
+
```
Comment (jamesej) [inline packages/web-app/src/hooks/useDaoProposal.tsx]: I'd say `useMemo` here is overkill as `generateCachedProposalId` would never be expensive to run, or is there some other reason to use `useMemo`? Not a big deal though, no need to change.

File: packages/web-app/src/hooks/useProposals.tsx [hunk 2]
```diff
@@ -86,8 +102,48 @@ export function useProposals(
...
+    return {
+      ...proposal,
+
+      result: {
+        ...proposal.result,
+        [voteValue]: proposal.result[voteValue] + 1,
+      },
+    };
+  }
+}
```
Comment (jamesej) [inline packages/web-app/src/hooks/useProposals.tsx]: This code is essentially the same as that in useDaoProposal, except for leaving out a couple of properties on the returned object. At some point would be nice to refactor this into a single separate function.

**Review by RakeshUP — APPROVED** [review]

**Review by jamesej — APPROVED** [review]
All looks fine

File: packages/web-app/src/hooks/useProposals.tsx (hunk 2 above)
Comment (Fabricevladimir) [reply packages/web-app/src/hooks/useProposals.tsx]: That's true. Good catch. I reckon I can get away with casting to the smaller type 🤔

**Review by jamesej — APPROVED** [review]
All good thanks

**Pull request merged**